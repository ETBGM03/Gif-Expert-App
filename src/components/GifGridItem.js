import React from "react";
import PropTypes from "prop-types";
import "animate.css";
const GifGridItem = ({ id, title, url }) => {
  return (
    <div
      className="card animate__animated animate__zoomIn animate__delay-1s"
      key={id}
    >
      <h2 className="title">{title}</h2>
      <img src={url} alt={title} />
    </div>
  );
};

GifGridItem.protoTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};

export default GifGridItem;
