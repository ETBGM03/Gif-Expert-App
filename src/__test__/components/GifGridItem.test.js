import React from "react";
import { shallow } from "enzyme";
import GifGridItem from "../../components/GifGridItem";

describe("Pruebas en <GifGridItem />", () => {
  const title = "Titulo del GIF";
  const url = "https://localhost/algo.png";
  const wrapper = shallow(<GifGridItem title={title} url={url} />);

  test("debe de mostrar el componente correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("debe de terner un h2 con el titulo", () => {
    const h2 = wrapper.find("h2");
    expect(h2.text().trim()).toBe(title);
  });

  test("la img debe tener dos atributos", () => {
    const img = wrapper.find("img");
    expect(img.prop("src")).toBe(url);
    expect(img.prop("alt")).toBe(title);
  });

  test("el div debe tener en una de sus clases animate__delay-1s", () => {
    const div = wrapper.find("div");
    expect(div.hasClass("animate__delay-1s")).toBe(true);
  });
});
