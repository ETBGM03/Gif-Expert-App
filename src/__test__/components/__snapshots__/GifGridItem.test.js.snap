// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> debe de mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__zoomIn animate__delay-1s"
>
  <h2
    className="title"
  >
    Titulo del GIF
  </h2>
  <img
    alt="Titulo del GIF"
    src="https://localhost/algo.png"
  />
</div>
`;
